src/app/services/actions/query-action-creator-util.ts

```diff
@@ -47,14 +52,76 @@ export function createAnonymousRequest(query: IQuery, proxyUrl: string) {
     ...sampleHeaders,
   };
 
-  const options: IRequestOptions = { method: query.selectedVerb, headers };
+  const options: IRequestOptions = {
+    method: query.selectedVerb,
+    headers,
+  };
   return { graphUrl, options };
 }
 
-export function authenticatedRequest(dispatch: Function, query: IQuery,
-  scopes: string[] = DEFAULT_USER_SCOPES.split(' ')) {
+export function authenticatedRequest(
+  dispatch: Function,
+  query: IQuery,
+  scopes: string[] = DEFAULT_USER_SCOPES.split(' ')
+) {
   dispatch(queryRunningStatus(true));
-  return makeRequest(query.selectedVerb, scopes)(query);
+  return makeGraphRequest(scopes)(query);
+}
+
+function createAuthenticatedRequest(
+  scopes: string[],
+  query: IQuery
+): GraphRequest {
+  const sampleHeaders: any = {};
+  sampleHeaders.SdkVersion = 'GraphExplorer/4.0';
+
+  if (query.sampleHeaders && query.sampleHeaders.length > 0) {
+    query.sampleHeaders.forEach((header) => {
+      sampleHeaders[header.name] = header.value;
+    });
+  }
+
+  const msalAuthOptions = new MSALAuthenticationProviderOptions(scopes);
+  const middlewareOptions = new AuthenticationHandlerOptions(
+    authProvider,
+    msalAuthOptions
+  );
+  const client = GraphClient.getInstance()
+    .api(encodeHashCharacters(query))
+    .middlewareOptions([middlewareOptions])
+    .headers(sampleHeaders)
+    .responseType(ResponseType.RAW);
+
+  return client;
+}
+
+export function makeGraphRequest(scopes: string[]): Function {
+  return async (query: IQuery) => {
+    let response;
+
+    const client = createAuthenticatedRequest(scopes, query);
```

Comment:
The function name `createAuthenticatedRequest` does not match with the variable name `client`. It can be made clearer by possibly renaming `client` to `request`

Reply:
I agree